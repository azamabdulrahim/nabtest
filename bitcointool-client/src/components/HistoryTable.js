import React, { Component } from "react";

class HistoryTable extends Component {
  render() {
    const { history } = this.props;

    let historyTable;

    if (history && history.length > 0) {
      historyTable = (
        <table className="table table-bordered">
          <tbody>
            <tr>
              <th>Currency</th>
              <th>Date</th>
              <th>Time</th>
              <th>Price</th>
            </tr>
            {history.map(data =>
              data.quotes.map(quote => (
                <tr>
                  <td>{data.currency}</td>
                  <td>{data.date}</td>
                  <td>{quote.time}</td>
                  <td>{quote.price}</td>
                </tr>
              ))
            )}
          </tbody>
        </table>
      );
    }

    return <div>{historyTable}</div>;
  }
}

export default HistoryTable;
