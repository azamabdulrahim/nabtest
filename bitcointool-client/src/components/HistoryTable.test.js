import React from "react";
import { shallow } from "enzyme";
import HistoryTable from "./HistoryTable";

describe("HistoryTable", () => {
  let wrapper;

  it("renders correctly with no data", () => {
    wrapper = shallow(<HistoryTable />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("table").length).toEqual(0);
  });

  it("renders correctly with data", () => {
    const history = [
      {
        currency: "BTC",
        date: "20180507",
        quotes: [
          {
            time: "1025",
            price: "2.03"
          },
          {
            time: "1200",
            price: "2.45"
          },
          {
            time: "1430",
            price: "2.30"
          }
        ]
      },
      {
        currency: "ETC",
        date: "20180507",
        quotes: [
          {
            time: "0915",
            price: "1.23"
          },
          {
            time: "1130",
            price: "1.45"
          }
        ]
      }
    ];

    wrapper = shallow(<HistoryTable history={history} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("table").length).toEqual(1);
    expect(wrapper.find("tr").length).toEqual(6);
    expect(
      wrapper.containsMatchingElement(
        <tr>
          <td>BTC</td>
          <td>20180507</td>
          <td>1025</td>
          <td>2.03</td>
        </tr>
      )
    ).toBeTruthy();
    expect(
      wrapper.containsMatchingElement(
        <tr>
          <td>ETC</td>
          <td>20180507</td>
          <td>1130</td>
          <td>1.45</td>
        </tr>
      )
    ).toBeTruthy();
  });
});
