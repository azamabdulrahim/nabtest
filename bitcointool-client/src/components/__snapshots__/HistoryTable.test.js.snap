// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryTable renders correctly with data 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <tbody>
      <tr>
        <th>
          Currency
        </th>
        <th>
          Date
        </th>
        <th>
          Time
        </th>
        <th>
          Price
        </th>
      </tr>
      <tr>
        <td>
          BTC
        </td>
        <td>
          20180507
        </td>
        <td>
          1025
        </td>
        <td>
          2.03
        </td>
      </tr>
      <tr>
        <td>
          BTC
        </td>
        <td>
          20180507
        </td>
        <td>
          1200
        </td>
        <td>
          2.45
        </td>
      </tr>
      <tr>
        <td>
          BTC
        </td>
        <td>
          20180507
        </td>
        <td>
          1430
        </td>
        <td>
          2.30
        </td>
      </tr>
      <tr>
        <td>
          ETC
        </td>
        <td>
          20180507
        </td>
        <td>
          0915
        </td>
        <td>
          1.23
        </td>
      </tr>
      <tr>
        <td>
          ETC
        </td>
        <td>
          20180507
        </td>
        <td>
          1130
        </td>
        <td>
          1.45
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`HistoryTable renders correctly with no data 1`] = `<div />`;
